wlauto/workloads/octaned8/__init__.py

```diff
@@ -0,0 +1,97 @@
+#    Copyright 2014-2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#pylint: disable=E1101,W0201
+
+import os
+
+from wlauto import Workload, Parameter, Executable
+from wlauto.exceptions import ConfigError
+
+class Octaned8(Workload):
+
+    name = 'octaned8'
+    description = """
+    Runs the Octane d8 benchmark.
+
+    This workload runs d8 binaries built from source and placed in the dependencies folder along
+    with test assets from https://github.com/chromium/octane which also need to be placed in an
+    assets folder within the dependencies folder.
+
+    Original source from::
+
+        https://github.com/v8/v8/wiki/D8%20on%20Android
+
+    """
+
+    parameters = [
+        Parameter('timeout', kind=int, default=120,
+                  description='Timeout, in seconds, for the script run time.'),
+    ]
+
+    supported_platforms = ['android']
+
+    executables = ['d8', 'natives_blob.bin', 'snapshot_blob.bin']
+
+    def setup(self, context):
```

Comment:
Assets deployment should probably be done inside `initialize()` rather than `setup()` as it only needs to be done once per run.